cloudinit/cmd/query.py

```diff
@@ -278,12 +278,22 @@ def handle_args(name, args):
         return 1
     if args.format:
         payload = "## template: jinja\n{fmt}".format(fmt=args.format)
-        rendered_payload = render_jinja_payload(
-            payload=payload,
-            payload_fn="query commandline",
-            instance_data=instance_data,
-            debug=True if args.debug else False,
-        )
+        try:
+            rendered_payload = render_jinja_payload(
+                payload=payload,
+                payload_fn="query commandline",
+                instance_data=instance_data,
+                debug=True if args.debug else False,
+            )
+        except Exception as e:
+            error_msg = (
+                "Failed to render templated data due to jinja parsing error: "
+                "{error}".format(
+                    error=str(e),
+                )
+            )
+            print(error_msg)
```

Comment:
Isn't writing to stdout necessary, though? Without doing so, the user just gets zero feedback. nothing. This would defeat the point of this ticket, which is to give better feedback when errors are caused specifically by malformed jinja templating. 

I don't see how else to avoid this? I don't like that the cloud init runtime just calls the render command and causes this to spit out a stdout message in the logs but i don't know how to make it so that it only gives the stdout message when render is invoked directly from the cli?

Reply:
Removing the print and keeping the LOG.error will write to stderr, see [1] and [2].

This script writes output to stdout using print in a successful path, and to stderr using LOG.error / LOG.warning to stderr.

The intention of this suggestion is to keep consistency and not break possible consumers of this command.

[1] [cloudinit/cmd/query.py#L265](https://github.com/canonical/cloud-init/blob/14b76c4448e817570699421b27e234d60cf40bbb/cloudinit/cmd/query.py#L265)
[2] [cloudinit/cmd/query.py#L267](https://github.com/canonical/cloud-init/blob/14b76c4448e817570699421b27e234d60cf40bbb/cloudinit/cmd/query.py#L267)